_posts/2018-11-30-grpc-stacks.md

```diff
@@ -0,0 +1,40 @@
+---
+layout: post
+title: gRPC Language Stacks
+published: false
+permalink: blog/grpc-stacks
+author: Carl Mastrangelo
+author-link: https://carlmastrangelo.com
+company: Google
+company-link: https://www.google.com
+---
+
+Here is a high level overview of the gRPC Stacks.  Each of the **10** default languages supported by gRPC has multiple layers, allowing you to customize what pieces you want in your application.
+
+<!--more-->
+
+
+There are three main stacks in gRPC: C-core, Go, and Java.  Most of the languages are thin wrappers on top of the [C-based](https://github.com/grpc/grpc/tree/v1.16.1/src/core) gRPC core library:
+
+<p><img src="https://grpc.io/img/grpc-core-stack.svg" alt="gRPC Core Stack" style="max-width: 800px" /></p>
+
+For example, a Python application calls into the generated Python stubs.  These call pass through interceptors, and into the wrapping library where the call is translated into C calls.  The gRPC C-core will encode the RPC as HTTP/2, optionally encrypt the data with TLS, and then pass write it to the network.
```

Comment:
These *calls* ...
...where the *calls are* ...

Reply:
then ~pass~ write

Reply:
Done.